src/util/dstring.h

```diff
@@ -158,6 +158,12 @@ inline size_t hash_string(const dstringt &s)
   return s.hash();
 }
 
+inline size_t const_hash_string(const dstringt &s)
+{
+  return hash_string(string_container.get_string(s.get_no()));
+}
```

Comment:
In line with what @martin-cs said: what exactly does this seek to achieve? In particular, this actually invokes the much more costly `std::string` version of `hash_string` !?